module/platform/webots/Webots/src/Webots.cpp

```diff
@@ -165,64 +165,69 @@ namespace module::platform::webots {
-                    // (method stolen from HardwareIO.cpp)
-                    // velocity = distance / time
-                    const float distance =
-                        utility::math::angle::difference(target.position,
-                                                         getDarwinServo(target.id, sensors).presentPosition);
-                    Clock::duration time = target.time - Clock::now();
-                    float velocity;
-                    if (time.count() > 0) {
-                        velocity = distance / static_cast<float>(time.count()) / static_cast<float>(Clock::period::den);
-                    }
-                    else {
-                        velocity = 0;
-                    }
-                    velocity_msg.name     = target.id;
-                    velocity_msg.velocity = velocity;
-                    to_send.motor_velocities.push_back(velocity_msg);
-
-                    MotorTorque torque_msg;
-                    torque_msg.name   = target.id;
-                    torque_msg.torque = target.torque;
-                    to_send.motor_torques.push_back(torque_msg);
-
-                    // MotorPID, only sending P gain. Set I and D to zero
-                    MotorPID motorpid_msg;
-                    motorpid_msg.name  = target.id;
-                    motorpid_msg.PID.X = static_cast<double>(target.gain);
-                    motorpid_msg.PID.Y = 0.0;
-                    motorpid_msg.PID.Z = 0.0;
-                    to_send.motor_pids.push_back(motorpid_msg);
+        on<Trigger<ServoTargets>, With<DarwinSensors>>().then([this](const ServoTargets& commands,
+                                                                     const DarwinSensors& sensors) {
+            message::platform::webots::ActuatorRequests to_send_next;
+
+            // Store each ServoTarget to send in the next lot
+            for (const auto& target : commands.targets) {
+                MotorPosition position_msg;
+                position_msg.name     = target.id;
+                position_msg.position = target.position;
+                to_send_next.motor_positions.push_back(position_msg);
+
+                MotorVelocity velocity_msg;
+                // We need to calculate the servo velocity to add to the velocity message
+                // (method stolen from HardwareIO.cpp)
+                // velocity = distance / time
+                const float distance =
+                    utility::math::angle::difference(target.position,
+                                                     getDarwinServo(target.id, sensors).present_position);
+
+                NUClear::clock::duration time = target.time - NUClear::clock::now();
+                float velocity;
+                if (time.count() > 0) {
+                    velocity =
+                        distance / static_cast<float>(time.count()) / static_cast<float>(NUClear::clock::period::den);
```

Comment:
Do we need to multiply by the numerator, to make sure nothing breaks if the numerator is not one?

Reply:
I don't understand what you mean

Reply:
a `std::ratio` has both a numerator and a denominator, to properly apply it you have to times by `num / den`.

Reply:
Oh, interesting. Both [`HardwareIO`](https://github.com/NUbots/NUbots/blob/master/module/platform/darwin/HardwareIO/src/HardwareIO.cpp#L381) and the [gazebo set up](https://github.com/NUbots/NUbots/pull/188/files#diff-6f602e793f7adecdd41295d32cee5df47594442833c48e2c77c497b2ec251a94R338) do it this way, so I suppose not?

Reply:
It just seems like code that will break if the numerator is not 1.

Reply:
This makes sense to me, and I'll explain my thinking with an example.
Say I went 2 units of distance in one unit of time => `time.count() == 1`.
Then, alternatively I go 2 units of distance in two units of time => `time.count() == 2`. 
This math works out so that if `time.count()` doubles, with the distance covered equal, you go half as fast, as you would expect to.

Reply:
Lets say that the tick ratio was `2/3` so after 2 seconds 3 ticks will have passed. Now if we multiply the number of ticks by `1/3` we get that `1` seconds have pass, when in really `2` seconds have passed.

Reply:
I think I see your point. What's your suggested solution?

Reply:
Basically multiply by the numerator, we have been lazy because we have always assumed the numerator is 1 :P